pomegranate/distributions/IndependentComponentsDistribution.pyx

```diff
@@ -253,10 +261,24 @@ cdef class IndependentComponentsDistribution(MultivariateDistribution):
 		X, weights = weight_set(X, weights)
 		n, d = X.shape
 
-		if callable(distributions):
-			distributions = [distributions.from_samples(X[:,i], weights) for i in range(d)]
-		else:
-			distributions = [distributions[i].from_samples(X[:,i], weights) for i in range(d)]
+		initialised_distributions = []
```

Comment:
I'm not sure I like all the additions. Maybe just change it to something like `distributions = [distributions.from_samples(X[:,i], weights, **params) for i in range(d)]` where params is {} normally or {'pseudocount': pseudocount} if that's passed in?